packages/dom-expressions/src/jsx.d.ts

```diff
@@ -57,13 +57,26 @@ export namespace JSX {
     $ServerOnly?: boolean;
   }
   interface Directives {}
+  interface DirectiveFunctions {}
```

Comment:
This could potentially restrict the types of directive functions allowed via an index signature:
```ts
    interface DirectiveFunctions {
      [x: string]: (el: Element, accessor: Accessor<any>) => void;
    }
```

Then later, filter out the index signature using `[Key in keyof DirectiveFunctions as string extends Key ? never : `use:${Key}`]?: ...`, or some other method. 

However there's no precedent for checking user-defined types elsewhere, so I'm not sure if it should be done.